ospfd/ospf_auth.c

```diff
@@ -222,22 +224,21 @@ static int ospf_auth_check_md5_digest(struct ospf_interface *oi,
 static int ospf_auth_make_md5_digest(struct ospf_interface *oi,
 				     struct ospf_packet *op, struct key *key)
 {
-	void *ibuf;
-	struct ospf_header *ospfh;
+	void *ibuf = STREAM_DATA(op->s);
+	struct ospf_header *ospfh = (struct ospf_header *)ibuf;
 	unsigned char digest[OSPF_AUTH_MD5_SIZE];
-	uint16_t length;
+	unsigned int length = ntohs(ospfh->length);
 #ifdef CRYPTO_OPENSSL
 	EVP_MD_CTX *ctx;
 #elif CRYPTO_INTERNAL
 	MD5_CTX ctx;
 #endif
 	char auth_key[OSPF_AUTH_MD5_SIZE + 1];
 
+	assert(length <= UINT32_MAX); /* for coverity's sake */
```

Comment:
Isn't this a 16 byte field?  Shouldn't we check that the length is less than UINT16_MAX and that the length is less than the length of the stream data?

Reply:
Yes it is a 16 byte field, that was my mistake. It's length should be lesser equal to the stream data. Will add these condition to assert

Reply:
I reverted `unsigned int` to `uint16_t`. I thought the coverity is complaining about passing `uint16_t` to `MD5Update` instead of `unsigned int`